app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +34,43 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def show_container(record, controller_name, display_name)
+    return if record_no_longer_exists?(record)
+    @gtl_url = "/show"
+    drop_breadcrumb({:name => display_name,
+                     :url  => "/#{controller_name}/show_list?page=#{@current_page}&refresh=y"},
+                    true)
+    case @display
+    when 'timeline'
+      show_timeline
+    when 'performance'
+      show_performance
+    end
+  end
+
+  def show_container_display(record, display, clazz, alt_controller_name = nil)
+    title = ui_lookup(:tables => display)
+    drop_breadcrumb(:name => _("%{name} (All %{title})") % {:name => record.name, :title => title},
+                    :url  => "/#{alt_controller_name || controller_name}/show/#{record.id}?display=#{@display}")
+    @view, @pages = get_view(clazz, :parent => record) # Get the records (into a view) and the paginator
+    @showtype = @display
+  end
+
+  def show_timeline
```

Comment:
this method seems to repeat itself in several places (not only middleware), any chance for unifying them?